### DISCUSSION TITLE
Keep built dependencies (images, css, etc) in relevant folders (Parcel v2)

### DISCUSSION THREAD

**Discussion by modulareverything** [parcel-bundler/parcel#5448] [Q&A]
I'm building a series of banners which need to be bundled together as standalone HTML files with their respective assets. I'm using Parcel to handle building them, and so I have structure like this

```
| -- src
  | -- banners
    | -- typeA
      | -- 300x300
        | -- 300x300.js
        | -- 300x300.scss
        | -- 300x300.jpg
        | -- index.scss
      | -- 500x600
        | -- 500x600.js
        | -- 500x600.scss
        | -- 500x600.jpg
        | -- index.scss
      | -- 750x400
        | -- 750x400.js
        | -- 750x400.scss
        | -- 750x400.jpg
        | -- index.scss
    | -- typeB
      | -- ...mimics typeA structure...
  | -- scss
    | -- global.scss
  | -- index.html
| package.json
```

The `index.html` file in the `src` root just pulls in all of the banners as iframes for demo purposes

This is generally working really well until I build. All of the files in the relevant folders are getting flattened down into the root `dist` directory.

**What I want is** to keep all of the dependencies in their correct folders, so I can individually share the HTML banners around with all of the bundled dependencies together (images, css, etc)

Is this possible?

**Comment by mischnic** [comment]
I don't completely understand your usecase, but this might work for your: https://github.com/mischnic/parcel-namer-preserve-structure

**Reply by modulareverything** [reply]
Just tried but maybe I didn't install it properly? I ran `npm install @mischnic/parcel-namer-preserve-structure` and created the `.parcelrc` file like this:

```
{
	"extends": "@parcel/config-default",
	"namers": ["@mischnic/parcel-namer-preserve-structure"]
}
```

Still getting the same output though:

```
dist/300x600.1fde72ab.css                                1.85 KB    584ms
dist/image.gif    178.09 KB    407ms
dist/300x600.3e461a78.js                                    46 B    584ms
dist/typeA/336x280/index.html
```

Am I missing a step?

**Reply by mischnic** [reply]
How are you invoking Parcel? 
And where does `typeA/336x280/index.html` come from, that isn't listed in your original post.

**Reply by modulareverything** [reply]
Via package.json scripts: `"build": "parcel build ./src/index.html"`
So just running `npm run build` in the console

I tried a variant `"build": "parcel build ./src/banners/**/index.html"` but no joy

> And where does typeA/336x280/index.html come from, that isn't listed in your original post.

I missed this out of my original post by mistake, but each folder (`300x300` etc) has its own `index.html` — hopefully that makes more sense to you now

**Reply by mischnic** [reply]
Seems to work for me: https://github.com/mischnic/parcel-issue-5448

**Reply by modulareverything** [reply]
I had to change `"parcel": "^2.0.0-beta.1",` to `"parcel": "^2.0.0-nightly.475",` which seems to do the trick

The only thing that doesn't seem to be working is the actual linking of the assets in the files. They're linking to `/filename.js` or whatever, which doesn't work. Changing it to `./filename.js` however works nicely

I can always change these manually but is there a better way?

**Reply by mischnic** [reply]
`--public-url ./`?

**Reply by modulareverything** [reply]
That strips the `/` but doesn't replace it with `./` as I thought it might

I think actually the problem is a bit more complex. I misread it before, but it's actually giving the asset the URL like this `/folder-name/asset-name.js`, so then it's looking for the file in `folder-name/folder-name/asset-name.js` when _actually_ the file is just in `folder-name/asset-name.js` — does that make sense?

So I think there needs to be some way of stripping out the `folder-name` entirely from the asset URL and replacing it with `./`
